src/main/java/com/shaft/tools/io/YAMLFileManager.java

```diff
@@ -0,0 +1,404 @@
+package com.shaft.tools.io;
+
+import com.shaft.tools.support.JavaActions;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.*;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+public class YAMLFileManager {
+    public final static String KEY_CONTAINS_LIST_REGEX = "^[-a-zA-Z\\d_!@#$%^&*()+=|\\\\/?><\"'{}~]*(\\[\\d+])+$";
+    public final static String NUMBER_IN_SQUARE_BRACKETS_REGEX = "\\[\\d+]";
+    public final static String SQUARE_BRACKETS_REGEX = "[\\[\\]]";
+    public final static String KEY_SEPARATOR_REGEX = "\\.";
+
+    private final Map<String, Object> data;
+    private final String filePath;
+
+    // Splitting the key to a list of keys using the dot as the splitter.
+    private final Function<String, List<String>> splitKey =
+            key -> Arrays.stream(key.split(KEY_SEPARATOR_REGEX)).toList();
+    // Checking if the key has an index for a list [index]
+    private final Predicate<String> keyHasList =
+            key -> key.matches(KEY_CONTAINS_LIST_REGEX);
+
+    /**
+     * Creating an instance of {@link YAMLFileManager}
+     *
+     * @param filePath target test data yaml file path
+     */
+    public YAMLFileManager(String filePath) {
```

Comment:
We should add a log message and attach the test data file to the report here in alignment with this method
https://github.com/ShaftHQ/SHAFT_ENGINE/blob/b90e8ff225776bc014f6f57680fc3bf67f646e24/src/main/java/com/shaft/tools/io/JSONFileManager.java#L27-L41

Reply:
Thanks 
I will work on these notes.

Reply:
I think I solved them.
Thanks